packages/react-events/src/dom/Tap.js

```diff
@@ -0,0 +1,753 @@
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:start';
+  const onTapStart = props.onTapStart;
+  if (onTapStart != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapStart);
+  }
+}
+
+function onChange(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const onTapChange = props.onTapChange;
+  if (onTapChange != null) {
+    const payload = state.isActive;
+    dispatchDiscreteEvent(context, payload, onTapChange);
+  }
+}
+
+function dispatchUpdate(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+) {
+  const type = 'tap:update';
+  const onTapUpdate = props.onTapUpdate;
+  if (onTapUpdate != null) {
+    const payload = {...state.gestureState, type};
+    dispatchUserBlockingEvent(context, payload, onTapUpdate);
+  }
+}
+
+function dispatchEnd(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:end';
+  const onTapEnd = props.onTapEnd;
+  if (onTapEnd != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapEnd);
+  }
+}
+
+function dispatchCancel(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:cancel';
+  const onTapCancel = props.onTapCancel;
+  if (onTapCancel != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapCancel);
+  }
+}
+
+function addRootEventTypes(
+  rootEventTypes: Array<string>,
+  context: ReactDOMResponderContext,
+  state: TapState,
+): void {
+  if (!state.rootEvents) {
+    state.rootEvents = rootEventTypes;
+    context.addRootEventTypes(state.rootEvents);
+  }
+}
+
+function removeRootEventTypes(
+  context: ReactDOMResponderContext,
+  state: TapState,
+): void {
+  if (state.rootEvents != null) {
+    context.removeRootEventTypes(state.rootEvents);
+    state.rootEvents = null;
+  }
+}
+
+function shouldActivate(event) {
+  const nativeEvent: any = event.nativeEvent;
+  const pointerType = event.pointerType;
+  const buttons = nativeEvent.buttons;
+  const isContextMenu = pointerType === 'mouse' && nativeEvent.ctrlKey && isMac;
+  const isValidButton =
+    buttons === buttonsEnum.primary || buttons === buttonsEnum.middle;
+
+  if (pointerType === 'touch' || (isValidButton && !isContextMenu)) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function isActivePointer(event, state) {
+  return (
+    state.pointerType === event.pointerType &&
+    state.pointerId === event.pointerId
+  );
+}
+
+function isModifiedTap(event) {
+  const nativeEvent: any = event.nativeEvent;
+  const {altKey, ctrlKey, metaKey, shiftKey} = nativeEvent;
+  return (
+    altKey === true || ctrlKey === true || metaKey === true || shiftKey === true
+  );
+}
+
+function createInitialState() {
```

Comment:
Should have a Flow return type to make sure we don't accidentally have property mismatches.